plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,648 @@
+This can occur if the owner of an input to an in-flight transaction is malicious and has signed a second transaction spending the same input.
+
+To account for this problem, we allow exits to be ambiguous about canonicity.
+Users can start MoreVP exits with the *assumption* that the referenced transaction is canonical.
+Other owners of inputs or outputs to the transaction can then join (or “piggyback”) the exit.
+We add cryptoeconomic mechanisms that determine whether the transaction is canonical and which of the inputs or outputs are unspent.
+The end result is that we can correctly determine which inputs or outputs should be paid out.
+
+
+### MoreVP Exit Protocol Specification
+
+#### Timeline
+
+The MoreVP exit protocol requires the use of a “challenge-response” mechanism, whereby users can submit a challenge but are subject to a response that invalidates the challenge.
+To give users enough time to respond to a challenge, the exit process is split into two “periods.” When challenges are subject to a response, we require that the challenges be submitted before the end of the first exit period and that responses be submitted before the end of the second.
+We define each period to have a length of half the minimum exit period (MEP).
+Currently, the MEP is set to 7 days, so each period has a length of 3.5 days.
+
+
+#### Starting the Exit
+
+For simplicity, we block deposit transactions from using the MoreVP exit protocol.
+
+Any user may initiate an exit by presenting a spend transaction and proving that the transaction is exitable.
+The user must submit a bond, `exit bond`, for starting this action.
+This bond is later used to cover the cost for other users to publish statements about the canonicity of the transaction in question.
+
+We provide several possible mechanisms that allow a user to prove a transaction is exitable.
+First, remember that deposit transactions are exitable by default.
+Two ways in which spend transactions can be proven exitable are as follows:
+1. The user may present the transaction along with each of the transactions that created the inputs to the transaction, a Merkle proof of inclusion for each of these transactions, and a signature from each input owner.
+The contract can then validate that these transactions are the correct ones, that they were included in the chain, that the signatures are correct, and that the exiting transaction is correctly formed.
+This proves the exitability of the transaction.
+2. The user may present the transaction along signatures the user claims to be valid.
+The contract can validate that the exiting transaction is correctly formed.
+Another user can challenge one of these signatures by presenting some transaction that created an input such that the true input owner did not sign the signature.
+In this case, the exit would be blocked entirely and the challenging user would receive `exit bond`.
+
+Option (1) checks that a transaction is exitable when the exit is started.
```

Comment:
I think we should leave a `TODO` self note to groom this part as soon as the option is chosen

Reply:
Would this be better as a markdown comment (hidden from view) or as a visible TODO?

Reply:
@kfichter hmmm, not sure. I'd say visible, so that the readers see where the doc is headed and that we don't forget too soon, that we've left a todo